zkevm-circuits/src/evm_circuit/util/common_gadget/tx_eip2930.rs

```diff
@@ -0,0 +1,134 @@
+use super::{CachedRegion, Cell};
+use crate::{
+    evm_circuit::util::{
+        constraint_builder::EVMConstraintBuilder, math_gadget::IsEqualGadget, select,
+    },
+    table::TxFieldTag,
+    util::Expr,
+    witness::Transaction,
+};
+use bus_mapping::circuit_input_builder::CopyDataType;
+use eth_types::{
+    geth_types::{access_list_size, TxType},
+    Field,
+};
+use halo2_proofs::{
+    circuit::Value,
+    plonk::{Error, Expression},
+};
+
+/// Transaction EIP-2930 gadget to handle optional access-list
+#[derive(Clone, Debug)]
+pub(crate) struct TxEip2930Gadget<F> {
+    is_eip2930_tx: IsEqualGadget<F>,
+    access_list_address_len: Cell<F>,
+    access_list_storage_key_len: Cell<F>,
+    access_list_gas_cost: Cell<F>,
+}
+
+impl<F: Field> TxEip2930Gadget<F> {
+    pub(crate) fn construct(
+        cb: &mut EVMConstraintBuilder<F>,
+        tx_id: Expression<F>,
+        tx_type: Expression<F>,
+    ) -> Self {
+        let is_eip2930_tx = IsEqualGadget::construct(cb, tx_type, (TxType::Eip2930 as u64).expr());
+
+        let [access_list_address_len, access_list_storage_key_len, access_list_gas_cost] = cb
+            .condition(is_eip2930_tx.expr(), |cb| {
+                let [address_len, storage_key_len, gas_cost] = [
+                    TxFieldTag::AccessListAddressesLen,
+                    TxFieldTag::AccessListStorageKeysLen,
+                    TxFieldTag::AccessListGasCost,
```

Comment:
you won't need this

Reply:
Yes, removed it in commit [6880378](https://github.com/scroll-tech/zkevm-circuits/pull/1045/commits/68803783b1447b8fefe2c1fabd626beae51b8a83), and calculate gas-cost as:
```
access_list_address_len.expr() * GasCost::ACCESS_LIST_PER_ADDRESS.expr()
    + access_list_storage_key_len.expr() * GasCost::ACCESS_LIST_PER_STORAGE_KEY.expr()
```